WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1743,6 +1748,150 @@ public boolean addMedia(Uri mediaUri) {
+            this.mWeakProgressDialog = new WeakReference<ProgressDialog>(progressDialog);;
+        }
+
+        public void setStopped(boolean stopped) {
+            isStopped = stopped;
+        }
+
+        @Override
+        public void onMediaStart() {
+
+        }
+
+        @Override
+        public void onMediaProgress(float progress) {
+
+        }
+
+        @Override
+        public void onMediaDone() {
+            ProgressDialog pd = mWeakProgressDialog.get();
+            dismissProgressDialog(pd);
+            if (isStopped || isFinishing()) {
+                return;
+            }
+            EditPostActivity.this.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    if (isStopped || isFinishing()) {
+                        return;
+                    }
+                    // Make sure the resulting file is smaller than the original
+                    long originalFileSize = FileUtils.length(mOriginalPath);
+                    long optimizedFileSize = FileUtils.length(mOutFilePath);
+                    String pathToUse = mOutFilePath;
+                    if (optimizedFileSize > originalFileSize) {
+                        AppLog.w(AppLog.T.MEDIA, "Optimized video is larger than original file "
+                                + optimizedFileSize + " > " + originalFileSize );
+                        pathToUse = mOriginalPath;
+                    }
+                    // Upload the file
+                    uploadFile(pathToUse);
+                }
+            });
+        }
+
+        private void uploadFile(String path) {
+            Uri uri = Uri.parse(path);
+            MediaModel media = queueFileForUpload(uri, getContentResolver().getType(uri));
+            MediaFile mediaFile = FluxCUtils.mediaFileFromMediaModel(media);
+            if (media != null) {
+                mEditorFragment.appendMediaFile(mediaFile, path, mImageLoader);
+            }
+        }
+
+        @Override
+        public void onMediaPause() {
+
+        }
+
+        @Override
+        public void onMediaStop() {
+            // This seems to be called called in 2 cases. Do not use to check if we've manually stopped the composer.
+            // 1. When the encoding is done without errors, before onMediaDone!
+            // 2. When we call 'stop' on the media composer.
+        }
+
+        @Override
+        public void onError(Exception exception) {
+            AppLog.e(AppLog.T.MEDIA, "Can't optimize the video", exception);
+            if (isStopped || isFinishing()) {
+                return;
+            }
+            ProgressDialog pd = mWeakProgressDialog.get();
+            dismissProgressDialog(pd);
+            Context context = mWeakContext.get();
+            if (context != null) {
+                ToastUtils.showToast(context, R.string.video_optimization_generic_error_message, Duration.LONG);
```

Comment:
I'm wondering if this toast is necessary. The user can't do anything about it, and I can't think of why they'd need to know this information.